events/tests/TESTS/events/queue/main.cpp

```diff
@@ -508,6 +508,82 @@ void static_events_queue_test()
     TEST_ASSERT_EQUAL(15, test4.counter);
 }
 
+static EventTest event_period;
+
+void event_period_tests()
+{
+    EventQueue queue;
+    
+    // Test a non periodic event ie dispatched only once
+    
+    Event<void(int)> event1(&queue, &event_period::f0);
+    event1.delay(100ms);
+    event1.period(Event::non_periodic);
+    event1.post()
+
+    queue.dispatch(800); 
+    
+    // Wait 1000ms and check the event execution status
+    wait_us(1000 * 1000)    
+    
+    // Event should only have been dispatched once and thus counter 
+    // should be 1
+    TEST_ASSERT_EQUAL(1, event_period.counter);
+
+    // Test an event with an invalid -ve period.
```

Comment:
`-ve` what does this mean? should it be -10ms?

Reply:
-ve = negative value for the period